src/main/kotlin/org/radarbase/redcap/integration/Integrator.kt

```diff
@@ -57,33 +57,31 @@ class Integrator(
             throw IllegalRequestException("Some of the required values were missing.")
         }
 
-        keys.add(IntegrationData.SUBJECT_ID_LABEL)
-        logger.info("Attribute Keys: {}", keys.toTypedArray())
-        val result: MutableMap<String, String> = try {
-            redCapIntegrator.pullFieldsFromRedcap(keys, recordId)
+        try {
+            attributes = redCapInfo.attributes?.associate { a -> (a.fieldName to redCapIntegrator.pullFieldFromRedcap(a.fieldName, recordId)) } ?: emptyMap()
+            redcapSubjectId = redCapIntegrator.pullFieldFromRedcap(SUJBECT_ID_KEY, recordId, enrolmentEvent)
```

Comment:
Can we not make a single request to get all the keys? 
Currently, this seems like added requests and will reduce performance (increasingly with each added attribute). This is especially important since we don't control all the redcap deployments, so can't be certain about the response rates.

Reply:
Oh okay I see. Yeah that’s true there could be potential performance issues. I made it separate because the results of the requests would be an array of the sets of attribute -> value mappings so I thought the multiple results might make the parsing more complicated. So should we just make the single request and process after?

Reply:
Yes would prefer to process after as Network I/O is unpredictable and slow.